src/main/java/com/google/cloud/dns/DnsKeyInfo.java

```diff
@@ -0,0 +1,377 @@
+        new Function<DnsKeyDigest, com.google.api.services.dns.model.DnsKeyDigest>() {
+          @Override
+          public com.google.api.services.dns.model.DnsKeyDigest apply(DnsKeyDigest dnsKeyDigest) {
+            return dnsKeyDigest.toPb();
+          }
+        };
+
+    private final String digest;
+    private final String type;
+
+    static class Builder {
+
+      private String digest;
+      private String type;
+
+      /**
+       * sets the base-16 encoded bytes of this digest. Suitable for use in a DS resource record.
+       */
+      Builder setDigest(String digest) {
+        this.digest = digest;
+        return this;
+      }
+
+      /** Specifies the algorithm used to calculate this digest. */
+      Builder setType(String type) {
+        this.type = type;
+        return this;
+      }
+
+      DnsKeyDigest build() {
+        return new DnsKeyDigest(this);
+      }
+    }
+
+    private DnsKeyDigest(Builder builder) {
+      this.digest = builder.digest;
+      this.type = builder.type;
+    }
+
+    /** Returns the base-16 encoded bytes of this digest. */
+    public String getDigest() {
+      return digest;
+    }
+
+    /** Returns the algorithm type of this digest. */
+    public String getType() {
+      return type;
+    }
+
+    static Builder newBuilder() {
+      return new Builder();
+    }
+
+    com.google.api.services.dns.model.DnsKeyDigest toPb() {
+      com.google.api.services.dns.model.DnsKeyDigest dnsKeyDigest =
+          new com.google.api.services.dns.model.DnsKeyDigest();
+      dnsKeyDigest.setDigest(digest);
+      dnsKeyDigest.setType(type);
+      return dnsKeyDigest;
+    }
+
+    static DnsKeyDigest fromPb(com.google.api.services.dns.model.DnsKeyDigest dnsKeyDigest) {
+      Builder builder = newBuilder();
+      if (dnsKeyDigest.getDigest() != null) {
+        builder.setDigest(dnsKeyDigest.getDigest());
+      }
+      if (dnsKeyDigest.getType() != null) {
+        builder.setType(dnsKeyDigest.getType());
+      }
+      return builder.build();
+    }
+
+    @Override
+    public boolean equals(Object other) {
+      return (other instanceof DnsKeyDigest) && this.toPb().equals(((DnsKeyDigest) other).toPb());
+    }
+
+    @Override
+    public int hashCode() {
+      return Objects.hash(digest, type);
+    }
+
+    @Override
+    public String toString() {
+      return MoreObjects.toStringHelper(this).add("digest", digest).add("type", type).toString();
+    }
+  }
+
+  static class Builder {
+    private String algorithm;
+    private String creationTime;
+    private String description;
+    private List<DnsKeyDigest> digests;
+    private String id;
+    private Boolean isActive;
+    private Long keyLength;
+    private Integer keyTag;
+    private String kind;
+    private String publicKey;
+    private String type;
+
+    Builder setAlgorithm(String algorithm) {
+      this.algorithm = algorithm;
+      return this;
+    }
+
+    Builder setCreationTime(String creationTime) {
```

Comment:
same here and below all are

Reply:
these are setter read only so user don't need to set the value client side